### DISCUSSION TITLE
How do we decide whether to promote a creator to a "source"?

### DISCUSSION THREAD

**Discussion by sarayourfriend** [WordPress/openverse#2711] [General]
Some GLAM or research institutions have their own "source" page, like NASA. How do we decide when to create such a source? There are heaps of "sources" that rely on Flickr, like Boston Public Library or even niche ones like [New South Wales grassy ecosystems](https://www.flickr.com/people/nswgrassyecosystems/).

How do we decide when to turn a creator into a "source"? Do we have a policy about this? Is it restricted to special-interest societies like NASA or a GLAM institution? Would something like the NSW grassy ecosystems page be a source?

From a search perspective, maybe it's not as important, but from a perspective of making these sources _discoverable_, I could see that being an interesting angle. For someone wanting to contribute to the commons, maybe if they've got pictures of NSW grassy ecosystems they'd like to know they can contribute to a place that will help them describe the images with accurate metadata?

Just looking for ideas of how we can think about this in the future. Places like Boston Public Library (as an example, other GLAM institutions too) fall in line with our existing patterns, but how do we actually make the decision to create these as "sources"? Is it possible for us to do so without needing to backport the data into the catalogue? What are the core effective differences between a "source" and a "highlighted creator"? Is there any difference, really? Is it possible for a _source_ to have multiple creators? Maybe I'm assuming that there is a 1:1 relationship between a creator of a provider and what we call a "source".

**Comment by obulat** [comment]
I don't think we have any general guidelines for adding "sources". We do have a DAG[^1] that checks all [Flickr Commons participating institutions](https://www.flickr.com/commons/institutions), and suggest adding ones that we don't already use as "source" (or "sub_provider" in the catalog) with at least 300 items with CC0, PDM or CC licenses. We also had a check for Europeana "sub_providers" that was retired.

For context, here is a list of all "sub_providers" in the catalog: 
https://github.com/WordPress/openverse/blob/6de944472d188e351af5058675af42f66f12256e/catalog/dags/common/loader/provider_details.py

> Just looking for ideas of how we can think about this in the future. Places like Boston Public Library (as an example, other GLAM institutions too) fall in line with our existing patterns, but how do we actually make the decision to create these as "sources"?

Does the current criterion sound good: _A notable (GLAM) creator that has more than 300 items can be extracted as a source_ ?

What are the core effective differences between a "source" and a "highlighted creator"? Is there any difference, really? 

> Is it possible for a source to have multiple creators? Maybe I'm assuming that there is a 1:1 relationship between a creator of a provider and what we call a "source".

When I first read this discussion, I thought that a "source" is not usually a creator, but an institution. Then I checked Flickr - and it seems that most of the "sources" are also marked as "creators". However, some Smithsonian sources do not have this 1:1 relationship between the creator and a source.  Here's one of such examples: https://openverse.org/image/61f31a70-5484-4233-95ab-9298e78ebee8?q=cat (the creator is probably extracted incorrectly, and also doesn't have a creator_url even though it is available on the source site).


[^1]: `catalog/dags/maintenance/flickr_audit_sub_provider_workflow.py`

**Reply by sarayourfriend** [reply]
Wow, thank you for that technical background! I didn't know we already identified potential sources like that.

> A notable (GLAM) creator that has more than 300 items can be extracted as a source

That sounds as good a policy as any to me.

Regarding the Smithsonian ones, I actually wanted to raise an issue about whether it makes sense to have those sources separated. It makes the "sources" list in the frontend filters so long! There's probably _something_ to the idea of being able to group certain sources together or otherwise organise the list in a way that doesn't cause an issue. Separate discussion though...

Regarding the DAG, do you know if we run it regularly or reference it? Maybe it's something for us to revisit on some cadence, either yearly or biannually?

**Reply by AetherUnbound** [reply]
Great questions, and thanks for the follow-up Olga! That DAG (`flickr_audit_sub_provider_workflow`) actually runs `@monthly` per our documentation: https://github.com/WordPress/openverse/blob/main/catalog/DAGs.md#maintenance

For maintainers with access, the direct Airflow URL for it is here: https://airflow.openverse.engineering/dags/flickr_audit_sub_provider_workflow/grid

So I think we're running it regularly enough to capture those cases!

As for the Smithsonian sources, I would definitely support grouping them together for the sake of the filter list length 😅 As you mention though, different discussion!

**Reply by sarayourfriend** [reply]
Can we add "check for new proposed sources" to the meetup agenda? Shouldn't take longer than a few minutes, hopefully? The goal is to make it an ongoing practice. Doing it at every meetup is an appropriate interval and the product is just to create issues to add the sources.

**Reply by AetherUnbound** [reply]
We didn't get a chance for this session during our meetup. What do you think would be involved in it? Going through the list of open issues to look at what sources have been proposed? Or going out to find new sources of CC images to make issues for? If it's the latter, do you have resources or suggestions for how to identify sources?

**Reply by sarayourfriend** [reply]
Sorry, no, I mean _sources_ not _providers_. The DAG evaluates potential sources in existing providers, right? The task would just be to go through the DAG's suggestions and flag ones to create issues for. As I understand it, the implementation of a source is just configuration in the area that Olga linked before? Maybe there's more, but it's not the same as a _provider_..

**Reply by AetherUnbound** [reply]
Oh, apologies! I think I might be confused because, in the Flickr DAG we have, the DAG itself will alert if there are new sub-providers which have been identified, but only for Flickr. It sounds like you might be suggesting going through our other providers as well for potential sources too, is that right?

**Reply by sarayourfriend** [reply]
If there are relevant providers then yes, but the key thing I am suggesting is to action on the DAGs alerts at least once a year, regardless of the underlying providers it evaluates.

**Comment by obulat** [comment]
While reviewing a PR for Europeana script, I realized that we need to promote many of the institutions there to sources. Europeana is an "integrator" that collects data from many GLAM institutions all over Europe, and many of them do fit the criteria above (more than 300 items with an open license or marked with CC0 or PDM).
However, if we do promote all of the Europeana sources, the image filters on the frontend will become unusably long. I wonder if we should have a separate project for outlining the source criteria and automating source extraction and updates, especially from the providers like Europeana, Smithsonian, Flickr or Wikimedia; and improving the way providers and sources are displayed on the frontend.

**Reply by sarayourfriend** [reply]
I would bet that https://github.com/WordPress/openverse/issues/753 would be a decent remedy to that by showing the source groups/categories/types above the sources list. I'm not sure if this would complicate things more than necessary, but we could also nest the sources under those groupings as a nested check-box tree (ex. (but not a suggested dependency!) https://github.com/riophae/vue-treeselect)

That might also be helpful if we wanted to retain the Smithsonian sources, it would just create a three-level deep classification (e.g., GLAM / Smithsonian / Smithsonian: Natural History).
